### PR TITLE
Refactor monitoring role. Add output to Kafka

### PR DESCRIPTION
**Pull request by tymofii-polekhin**
-- roles/collectd/templates/collectd.conf.j2
Add percent output to cpu and memory metrics

-- roles/docker/files/dockerplugin.db
Update to latest version with percentage metrics

-- roles/docker/files/dockerplugin.py
Update to latest version. Add workaround with long containers name

-- roles/logstash/defaults/main.yml
Change the docker container to newer version with Kafka output support

-- roles/logstash/templates/logstash-service.j2
Mount volumes based on leader/follower role

-- roles/logstash/templates/logstash.conf.j2
Add support for output Kafka. Differentiate logs by type

### CODE REVIEW

File: roles/logstash/defaults/main.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
-logstash_image: ciscocloud/logstash
-logstash_image_tag: 0.2
+# Docker logstash container image
+logstash_image: tpolekhin/docker-logstash
```
Comment (stevendborrelli) [inline roles/logstash/defaults/main.yml]: Can we upgrade the ciscocloud image to this version?

Comment (tymofii-polekhin) [inline roles/logstash/defaults/main.yml]: Sure, will do and update PR

**Comment by tymofii-polekhin** [comment]
@stevendborrelli can you please review and merge?

**Comment by ryane** [comment]
I'll work on this today

**Comment by tymofii-polekhin** [comment]
@ryane thank you!

**Comment by ryane** [comment]
@tymofii-polekhin I am getting the following error

```
fatal: [resching-gce-worker-002] => {'msg': "AnsibleUndefinedVariable: One or more undefined variables: 'mesos_mode' is undefined", 'failed': True}
```

It looks like logstash is now dependent on the mesos role? Did you make changes to the terraform.yml to change when the logstash is applied?

**Comment by eirslett** [comment]
Maybe mesos_mode should be determined in terms of mantl_role? and mantl_role could be either "control" or "something else" (typically, "worker", but in the future there may be more worker roles)

**Comment by stevendborrelli** [comment]
Yes, doesn't the host already have a role= set?

**Comment by ryane** [comment]
yeah, good points. at this point, all of the `mesos_mode` conditionals could probably be replaced by testing against `role=control` and `role=worker`

**Comment by tymofii-polekhin** [comment]
@ryane @stevendborrelli fixed, tested and committed

**Comment by ryane** [comment]
This is looking good. I am able to see the updated cpu, memory, and docker metrics coming in. The docker metrics look ok. All of the mesos-* logs seem to be forwarding correctly. 

One thing I am not seeing is the `marathon-logs` type. @tymofii-polekhin can you explain how that is supposed to work?

**Comment by ryane** [comment]
I had to make a small change to fix the Kafka output example. It works now:

```
$ sudo docker run --rm -it ryane/kafkacat -b broker-0.service.consul:4000 -t logstash
% Auto-selecting Consumer mode (use -P or -C to override)
{"message":"I1111 15:33:03.076865  5078 slave.cpp:2671] Handling status update TASK_RUNNING (UUID: b1881d9a-0455-42d3-b976-676466c3472f) for task elasticsearch_resching-gce-worker-003_20151111T134213.609Z of framework 20151111-122647-117440522-15050-2141-0002 from executor(1)@10.0.0.2:48666","@version":"1","@timestamp":"2015-11-11T15:33:03.366Z","host":"resching-gce-worker-003","path":"/logs/mesos/mesos-slave.INFO","type":"mesos-slave-logs"}
{"message":"I1111 15:33:03.078296  5078 status_update_manager.cpp:322] Received status update TASK_RUNNING (UUID: b1881d9a-0455-42d3-b976-676466c3472f) for task elasticsearch_resching-gce-worker-003_20151111T134213.609Z of framework 20151111-122647-117440522-15050-2141-0002","@version":"1","@timestamp":"2015-11-11T15:33:03.368Z","host":"resching-gce-worker-003","path":"/logs/mesos/mesos-slave.INFO","type":"mesos-slave-logs"}
{"message":"I1111 15:33:03.079560  5078 status_update_manager.cpp:826] Checkpointing UPDATE for status update TASK_RUNNING (UUID: b1881d9a-0455-42d3-b976-676466c3472f) for task elasticsearch_resching-gce-worker-003_20151111T134213.609Z of framework 20151111-122647-117440522-15050-2141-0002","@version":"1","@timestamp":"2015-11-11T15:33:03.368Z","host":"resching-gce-worker-003","path":"/logs/mesos/mesos-slave.INFO","type":"mesos-slave-logs"}
{"message":"I1111 15:33:03.083745  5077 slave.cpp:2926] Forwarding the update TASK_RUNNING (UUID: b1881d9a-0455-42d3-b976-676466c3472f) for task elasticsearch_resching-gce-worker-003_20151111T134213.609Z of framework 20151111-122647-117440522-15050-2141-0002 to master@10.0.0.7:15050","@version":"1","@timestamp":"2015-11-11T15:33:03.369Z","host":"resching-gce-worker-003","path":"/logs/mesos/mesos-slave.INFO","type":"mesos-slave-logs"}
...
```

The only outstanding issue is the `marathon-logs` configuration. As far as I can tell, that is not doing anything at this point

**Comment by tymofii-polekhin** [comment]
@ryane the `marathon-logs` requires changing the marathon configuration to use `http-endpoint` to forward all deployment messages to the listening logstash.
Like this:
```
--event_subscriber http_callback --http_endpoints http://localhost:5678
```

Im not sure that this needs to be the part of this PR

**Comment by ryane** [comment]
I see, thanks for the clarification. This can also be enabled dynamically then by posting `http://localhost:5678` to the `/v2/eventSubscriptions` marathon api.

**Comment by tymofii-polekhin** [comment]
@ryane yes, but first we need to enable `--event_subscriber http_callback`

**Comment by ryane** [comment]
it's already enabled: https://github.com/CiscoCloud/microservices-infrastructure/blob/master/roles/marathon/tasks/conf.yml#L46

we currently register a callback for marathon-consul: https://github.com/CiscoCloud/microservices-infrastructure/blob/master/roles/marathon/tasks/jobs.yml#L36

**Comment by ryane** [comment]
merged, thanks

**Pull request merged**